rfcs/require_webdriver_bidi.md

```diff
@@ -0,0 +1,114 @@
+# RFC 212: Introduce `require_webdriver_bidi` metadata tag
+
+## Summary
+
+To facilitate the gradual enablement of WebDriver BiDi transports, this RFC proposes the introduction of a new metadata tag, `require_webdriver_bidi` in html and js files in WPT tests. This metadata would allow for a gradual transition from [WebDriverProtocol](https://github.com/web-platform-tests/wpt/blob/9c76757f5332678f9952f6ccb3824f62d30eca1f/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L644) to [WebDriverBidiProtocol](https://github.com/web-platform-tests/wpt/blob/9c76757f5332678f9952f6ccb3824f62d30eca1f/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L736), ensuring that implementations can opt-in into WebDriver BiDi gradually in case there are implementation blockers like in Chrome's case. [Prototype](https://github.com/web-platform-tests/wpt/pull/48622).
+
+## Background
+
+[RFC 185: Add WebDriver BiDi support to testdriver.js](https://github.com/web-platform-tests/rfcs/blob/master/rfcs/testdriver_bidi.md) discusses incorporating BiDi functionality into `testdriver.js`. This entails migrating from [WebDriverProtocol](https://github.com/web-platform-tests/wpt/blob/9c76757f5332678f9952f6ccb3824f62d30eca1f/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L644) to [WebDriverBidiProtocol](https://github.com/web-platform-tests/wpt/blob/9c76757f5332678f9952f6ccb3824f62d30eca1f/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L736). However, this switch may cause regression in certain tests that don't utilize the testdriver BiDi API due to unintended side effects resulting from the transport change. Nonetheless, all these regressions need to be resolved eventually. Implementing a flag will enable immediate provision of the testdriver BiDi API to tests, eliminating the need for test authors to wait for the migration of all previous tests to `WebDriverBidiProtocol`.
```

Comment:
> This feels like it presupposes migrating everything over to `WebDriverBiDiProtocol`, instead of continuing with the status quo of some being in `WebDriverProtocol` and others BiDi-exclusive being in `WebDriverBiDiProtocol`?

I guess migrating everything over to `WebDriverBiDiProtocol` can happen at certain point, but not in the nearest future. As long as we don't have consensus on that yet, let's consider we are going to support both.

> That wasn't in the previous RFC as far as I can see, and that is a much larger (and more breaking change) than we've previously discussed. It is highly likely that there will be a number of vendors with no BiDi implementation running wptrunner for a long time into the future, thus migrating things over to requiring BiDi when they can be implemented via Classic seems problematic.

This document does not restrict usage of `WebDriverProtocol` and such a migration (BiDi by default) should be discussed and planned separately.

Reply:
The wording is fixed. We stick to the #185 proposal, and do not propose to retire `WebDriverProtocol`.